avocado/plugins/xunit.py

```diff
@@ -63,8 +64,11 @@ def _create_testcase_element(self, document, state):
         name = state.get("name")
         if isinstance(name, TestID):
             testcase.setAttribute("name", self._escape_attr(name.name))
+            file_path, _ = reference_split(name.name)
         else:
             testcase.setAttribute("name", self._get_attr(state, "name"))
+            file_path, _ = reference_split(self._get_attr(state, "name"))
+        testcase.setAttribute("file", self._escape_attr(file_path))
```

Comment:
Can you provide some pointers about this (meaning `file`) being the actual attribute name?  Helpful pointers would include:

* Gitlab documentation or code describing where the `Filename` column comes from
* JUnit schema (updates?) showing the `file` attribute 

I've looked at a few pointers such as:
* https://www.ibm.com/docs/en/developer-for-zos/14.1?topic=formats-junit-xml-format
* https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd

And I failed to see the reference to the `file` attribute.

Also, looking at a [current test report](https://gitlab.com/qemu-project/qemu/-/pipelines/723305724/test_report) from GitLab, I failed to find any test with `Filename` data.

Reply:
> Can you provide some pointers about this (meaning `file`) being the actual attribute name? Helpful pointers would include:
> 
>     * Gitlab documentation or code describing where the `Filename` column comes from
> 
>     * JUnit schema (updates?) showing the `file` attribute
> 
> 
> I've looked at a few pointers such as:
> 
>     * https://www.ibm.com/docs/en/developer-for-zos/14.1?topic=formats-junit-xml-format
> 
>     * https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd
> 

Hi @clebergnu, you are right this is not part of the IBM's standard, and it is also missing in GitLab documentation. There is a GitLab [issue](https://gitlab.com/gitlab-org/gitlab/-/issues/299086) about it. But you can find the definitions in the [source code](https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/parsers/test/junit.rb). 

IMO it is still make sense to supported because it is working in github even it is not documented.

> 
> And I failed to see the reference to the `file` attribute.
> 
> Also, looking at a [current test report](https://gitlab.com/qemu-project/qemu/-/pipelines/723305724/test_report) from GitLab, I failed to find any test with `Filename` data.

You will find that the `Filename` data is missing in your example [here](https://gitlab.com/qemu-project/qemu/-/pipelines/723305724/test_report)